rep-2004.rst

```diff
@@ -0,0 +1,527 @@
+Refer to the detailed description of the requirements in the Quality Level 1 section above for more information.
+
+Quality Level 4
+^^^^^^^^^^^^^^^
+
+These are demos, tutorials, or experiments.
+They don't have strict requirements, but are not excluded from having good documentation or tests.
+For example, this might be a tutorial package which is not intended for reuse but has excellent documentation because it serves primarily as an example to others.
+
+Package Requirements
+~~~~~~~~~~~~~~~~~~~~
+
+*Note: bullets below that start with [ROS Core], will be the prescription for what we do in the core packages in order to meet the associated requirements*
+
+Requirements to be considered a 'Level 4' package:
+
+* Version Policy:
+
+  * No requirements, but having a policy is still recommended (e.g. ``semver``), even if the version is not yet stable (e.g. >= 1.0.0 for ``semver``)
+
+* Change Control Process:
+
+  * No explicit change control process required, but still recommended
+
+* Documentation:
+
+  * Must have a declared license or set of licenses
+  * Must have a copyright statement in each source file
+  * [ROS Core]:
+
+    * Must have automated checks for copyright statements and licenses
+    * Must use the Apache 2.0 license, unless the package has an existing permissive license (e.g. rviz uses three-clause BSD)
+
+* Testing:
+
+  * No explicit testing requirements, though covering some if not all of the features with tests is recommended
+
+* Dependencies:
+
+  * No restrictions
+
+* Platform Support:
+
+  * May support all tier 1 platforms for ROS 2, as defined in `REP-2000 <https://www.ros.org/reps/rep-2000.html#support-tiers>`_
+
+Any package that does not claim to be 'Level 3' or higher is automatically 'Level 4'.
+Refer to the detailed description of the requirements in the Quality Level 1 section above for more information.
+
+Quality Level 5
+^^^^^^^^^^^^^^^
+
+Packages in this category simply do not meet even the 'Level 4' requirements, and for that reason should not be used.
+The rationale being that all packages should have at least a declare license or licenses and should include copyright statements in each file.
+
+Repository Organization
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Since these categories are applied on a per package basis, and since there may be more than one package per source repository, it's recommended that the strictest set of policies apply to the whole repository.
+This is recommended, rather than trying to mix processes depending on which packages are changed in a given change request (pull request or merge request, etc.).
+If this is too onerous, then it's recommended to split lower quality packages out into a separate repository.
+
+
+Motivation
+==========
+
+TBD
+
+
+Rationale
+=========
+
+TBD
+
+
+Backwards Compatibility
+=======================
+
+Since there are no preceeding standards in this space, at least in the ROS community, of which we are aware, there are no backwards compatibility concerns.
```

Comment:
It may not have been as rigorously standardised in a REP, but at least in ROS-Industrial we've had the [Software Status indicators](http://wiki.ros.org/Industrial/Software_Status) for quite a while now.

These are not a bw-compatibility concern, but I believe they do qualify as existing work/prior art.

Reply:
Ah thanks, I looked briefly for this (I remembered something from ROS-I) but couldn't find it (I think I was looking for the 'quality' keyword). My mistake. I'll include it here, but as you say I don't think they're incompatible or affect backwards compatibility. I'll also read up on this wiki page and see if anything should be incorporated into this REP from it.